### PR TITLE
LightFlare functionality

### PR DESCRIPTION
**Pull request by IAmAllOfYou**

### CODE REVIEW

File: Assets/FlareSpawner.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class FlareSpawner : MonoBehaviour {
+
```
Comment (stellarxo) [inline Assets/FlareSpawner.cs]: { goes on its own line

File: Assets/FlareSpawner.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    [Tooltip("Cost to use your flare")]
+    public float flareEnergyCost;
+    [Tooltip("Percentage of energy needed to use flare. 1 = 100%")]
+    public float flareCostPercentage;
+
+    private float timer;
+    private LightSource lightSource;
+	// Use this for initialization
+	void Start () {
+        timer = cooldownTime;
```
Comment (stellarxo) [inline Assets/FlareSpawner.cs]: should be:

void Start() // no space between function name and brackets
{ // on its own line

File: Assets/FlareSpawner.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            {
+                if(lightSource.LightEnergy.CurrentEnergy > flareEnergyCost * flareCostPercentage)
+                {
+                    Instantiate(flareObject, flareSpawnObject.position, flareSpawnObject.rotation);
+                    lightSource.LightEnergy.Deplete(flareEnergyCost);
+                    timer = 0.0f;
+                }
+            }
+        }
+	}
```
Comment (stellarxo) [inline Assets/FlareSpawner.cs]: what is this spacing here

File: Assets/Scripts/Player/PlayerFlare.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        lightObject = gameObject.GetComponentInChildren<Light>();
+        flareLens = lightObject.GetComponent<LensFlare>();
+
+        rigidbody.velocity = transform.right * speed;
+        Destroy(gameObject, destroyTime);
+    }
+	
+	// Update is called once per frame
+	void Update()
+    {
```
Comment (stellarxo) [inline Assets/Scripts/Player/PlayerFlare.cs]: spacing here is weird

File: Assets/Scripts/Player/PlayerFlare.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        if ((timer += Time.deltaTime) > timeBeforeDecay)
+        {
+            Debug.Log(Time.deltaTime.ToString());
+            lightObject.intensity -= Time.deltaTime * lightReduction;
+            flareLens.brightness -= Time.deltaTime * fadeSpeed;
+            timer = 0.0f;
+            timeBeforeDecay = decayRateTime;
+        }
+        
+	}
```
Comment (stellarxo) [inline Assets/Scripts/Player/PlayerFlare.cs]: what are tabs

**Comment by stellarxo** [comment]
I would remove the scene that is 'just for you'

fix what happens with collisions/walls

and for the next iteration, work on flare doing a better "bounce" off of walls

**Comment by alisterdev** [comment]
@stellarxo this looks fine to me, please double check and merge

**Pull request merged**